integration/spark2/src/main/scala/org/apache/carbondata/store/SparkCarbonStore.scala

```diff
@@ -110,8 +110,16 @@ class SparkCarbonStore extends MetaCachedCarbonStore {
   }
 
   def startSearchMode(): Unit = {
+    LOG.info("Starting search mode master")
     master = new Master(session.sparkContext.getConf)
     master.startService()
+    var count = 0
+    while (master.isStarted().compareAndSet(false, false) && count < 5000) {
+      LOG.info(s"Waiting search mode master to start, retrying $count times")
+      Thread.sleep(10)
+      count = count + 1;
+    }
+    LOG.info("Search mode master started")
```

Comment:
Need one more if check, if (count == 5000), should throw exception because it is failed to start master

Reply:
ok